File: patch_pvp.js
const fs = require('fs');

let pvp = fs.readFileSync('src/routes/pvp.ts', 'utf8');

pvp = pvp.replace(
  "import { getQuestionsForBuilding } from '../data/questionsData.js';",
  "import { drizzleClient as db } from '../db/drizzle_client.js';\nimport { question } from '../data/model/question.js';\nimport { sql } from 'drizzle-orm';"
);

pvp = pvp.replace(
  "function getPvPQuestions(subject: string, grade: string = 'Class 10', curriculum: string = 'CBSE'): MCQuestion[] {",
  "async function getPvPQuestions(subject: string, grade: string = 'Class 10', curriculum: string = 'CBSE'): Promise<any[]> {"
);

const oldFuncBody = `
  try {
    const fallback = getQuestionsForBuilding('arena', subject);
    if (fallback && Array.isArray(fallback.questions) && fallback.questions.length > 0) {
      return [...fallback.questions].sort(() => 0.5 - Math.random()).slice(0, 5);
    }
  } catch (err) {
    console.error('PvP Question Fetch Error', err);
  }
  return [];
`;

const newFuncBody = `
  try {
    const rows = await db.select().from(question).orderBy(sql\`RANDOM()\`).limit(5);
    return rows.map((r: any) => ({
      id: r.id,
      question: r.question,
      options: ['Option A', 'Option B', 'Option C', 'Option D'], // TODO: Fetch from option table
      correct_answer: 'Option A', // TODO: Identify correct option
      explanation: 'Explanation',
    }));
  } catch (err) {
    console.error('PvP Question Fetch Error', err);
  }
  return [];
`;

// It's too complex to string replace the function body reliably, let's just rewrite the function

File: patch_pvp.py
import re

with open('src/routes/pvp.ts', 'r') as f:
    data = f.read()

# Replace imports
data = data.replace(
    "import { getQuestionsForBuilding } from '../data/questionsData.js';",
    "import { drizzleClient as db } from '../db/drizzle_client.js';\nimport { question } from '../data/model/question.js';\nimport { option } from '../data/model/option.js';\nimport { sql, eq } from 'drizzle-orm';"
)

pattern = r"function getPvPQuestions.*?return \[\];\n\}"
replacement = """async function getPvPQuestions(subject: string, grade: string = 'Class 10', curriculum: string = 'CBSE'): Promise<MCQuestion[]> {
  try {
    const randomQuestions = await db.select().from(question).orderBy(sql`RANDOM()`).limit(5);
    const formattedQuestions = [];
    for (const q of randomQuestions) {
      const opts = await db.select().from(option).where(eq(option.question_id, q.id));
      const correctAnswer = opts.find((o: any) => o.is_correct)?.option || '';
      formattedQuestions.push({
        id: q.id,
        question: q.question,
        options: opts.map((o: any) => o.option),
        correct_answer: correctAnswer,
        explanation: 'Keep learning!',
      });
    }
    return formattedQuestions as any;
  } catch (err) {
    console.error('PvP Question Fetch Error', err);
  }
  return [];
}"""

data = re.sub(pattern, replacement, data, flags=re.DOTALL)

with open('src/routes/pvp.ts', 'w') as f:
    f.write(data)
